executor/slow_query.go

```diff
@@ -866,7 +1029,14 @@ func (e *slowQueryRetriever) getAllFiles(sctx sessionctx.Context, logFilePath st
 			return handleErr(err)
 		}
 		end := types.NewTime(types.FromGoTime(fileEndTime), mysql.TypeDatetime, types.MaxFsp)
-		if end.Compare(e.checker.startTime) < 0 {
+		inTimeRanges = false
+		for _, tr := range e.checker.timeRanges {
+			if !(start.Compare(tr.endTime) > 0 || end.Compare(tr.startTime) < 0) {
+				inTimeRanges = true
+				break
+			}
+		}
+		if !inTimeRanges {
```

Comment:
`inTimeRanges ` variable here is confused, line 1022 is `if inTimeRanges `, but here is `if ! inTimeRanges`